### PR TITLE
Return address and type of imported key in z_importkey

### LINKED ISSUE #3748: Provide a way to obtain the address of a key imported by z_importkey
Reported by `@mariodo` [on community chat](https://chat.zcashcommunity.com/channel/zcash?msg=Zbta3ip3oa7eyFDdZ). The simplest approach is probably to have `z_importkey` return the address of the imported key.

### PR DESCRIPTION
**Pull request by oxarbitrage**
Fixes #3748.

### CODE REVIEW

**Comment by oxarbitrage** [comment]
I made this code last week without noticing the issue already had a PR. Please feel free to close it in favour of https://github.com/zcash/zcash/pull/3754

**Comment by oxarbitrage** [comment]
I ported the visitor from #3754 with some small modifications to address https://github.com/zcash/zcash/pull/3754#discussion_r277639467
Also ported the `rpc_wallet_z_importkey_paymentaddress` test case. 

My apologies @dagurval for taking over your code. I think now this one haves the best of the 2 so #3754 can be closed.

File: qa/rpc-tests/wallet_sapling.py [hunk 1]
```diff
@@ -136,11 +136,13 @@ def run_test(self):
 
         # Verify importing a spending key will update the nullifiers and witnesses correctly
         sk0 = self.nodes[0].z_exportkey(saplingAddr0)
-        self.nodes[2].z_importkey(sk0, "yes")
-        assert_equal(self.nodes[2].z_getbalance(saplingAddr0), Decimal('10'))
+        saplingAddr0 = self.nodes[2].z_importkey(sk0, "yes")
```
Comment (str4d) [inline qa/rpc-tests/wallet_sapling.py]: This shadows the original `saplingAddr0`, functionally altering it from an address to a tuple containing an address, which is likely to confuse future people modifying this test. Use a different name for the returned tuple.

File: qa/rpc-tests/wallet_sapling.py [hunk 2]
```diff
@@ -136,11 +136,13 @@ def run_test(self):
...
+        assert_equal(saplingAddr0["type"], "sapling")
+        assert_equal(self.nodes[2].z_getbalance(saplingAddr0["address"]), Decimal('10'))
         sk1 = self.nodes[1].z_exportkey(saplingAddr1)
-        self.nodes[2].z_importkey(sk1, "yes")
-        assert_equal(self.nodes[2].z_getbalance(saplingAddr1), Decimal('5'))
+        saplingAddr1 = self.nodes[2].z_importkey(sk1, "yes")
```
Comment (str4d) [inline qa/rpc-tests/wallet_sapling.py]: Ditto.

**Comment by oxarbitrage** [comment]
Fixed comments, rebased and squashed everything in just 1 commit co authored.

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**